src/files/actions/approveAction.js

```diff
@@ -0,0 +1,31 @@
+import CheckCircleSvgIcon from '@mdi/svg/svg/check-circle.svg'
+import { Permission, FileAction } from '@nextcloud/files'
+import { states } from '../../states.js'
+import { onApproveAction } from '../helpers.js'
+
+export const approveAction = new FileAction({
+	id: 'approval-approve',
+	displayName: (nodes) => {
+		return t('approval', 'Approve')
+	},
+	enabled(nodes, view) {
+		return !OCA.Approval.actionIgnoreLists.includes(view.id)
+			&& !nodes.some(({ permissions }) => (permissions & Permission.READ) === 0)
+			&& nodes.some(node => node.attributes['approval-state'] === states.APPROVABLE)
+		// && nodes.every(({ type }) => type === FileType.File)
+		// && nodes.every(({ mime }) => mime === 'application/some+type')
+	},
+	iconSvgInline: () => CheckCircleSvgIcon,
+	order: 0,
+	async exec(node) {
+		await onApproveAction(node)
+		return null
+	},
+	async execBatch(nodes) {
+		const promises = nodes
+			.filter(node => node.attributes['approval-state'] === states.APPROVABLE)
+			.map(onApproveAction)
+		const results = await Promise.allSettled(promises)
+		return results.map(promise => null)
```

Comment:
You're still missing the error handling.
If you approve a selection, and some items fail, you want to keep the failed elements selected.
You're losing that information here

Reply:
My goal was to avoid the standard showError/showSuccess message. I would like to keep the failed elements selected but avoid the messages because I already produce custom ones.

Anyway, error handling for multiple selection approve and reject is now better and we have the additional standard message. Fine by me. Or is there a way to avoid it?

Single selection (exec for approve, reject, request and inline) are still silent.

Reply:
Ok, now no more notification for each file in multiple selection.